### PR TITLE
[WIP] [Promotion] Apply promotion for users who have ordered in the last X months

### PR DESCRIPTION
**Pull request by jblanchon**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/messages.fr.yml [hunk 1]
```diff
@@ -34,7 +34,20 @@ sylius:
         after: Utilisateur inscrit après cette période ?
       shipping_country_configuration:
         country: Pays
-      type: Type
+      order_loyalty_configuration:
```
Comment (tristanbes) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/messages.fr.yml]: I think the translation process takes is different on Sylius, keep an eye on #3317 & https://github.com/Sylius/Sylius/pull/3316#issuecomment-142021207

File: src/Sylius/Component/Core/Promotion/Checker/OrderLoyaltyRuleChecker.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        if (null === $customer = $subject->getCustomer()) {
+            return false;
+        }
+
+        $date = new \DateTime(sprintf('%d %s ago', $configuration['nth'], $configuration['unit']));
+        $count  = $this->orderRepository->countByCustomerPaymentStateFromDate($customer, PaymentInterface::STATE_COMPLETED, $date, $configuration['after']);
+
+        if($configuration["equal"] == "up") {
```
Comment (tristanbes) [inline src/Sylius/Component/Core/Promotion/Checker/OrderLoyaltyRuleChecker.php]: PSR invalid, missing whitespace after `if`

File: src/Sylius/Component/Core/Promotion/Checker/OrderLoyaltyRuleChecker.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            if ($count >= $configuration['orderNth']) {
+                return true;
+            }
+        }else{
```
Comment (tristanbes) [inline src/Sylius/Component/Core/Promotion/Checker/OrderLoyaltyRuleChecker.php]: idem

File: src/Sylius/Component/Core/spec/Promotion/Checker/OrderLoyaltyRuleCheckerSpec.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Checker;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Resource\Model\TimestampableInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+/**
```
Comment (tristanbes) [inline src/Sylius/Component/Core/spec/Promotion/Checker/OrderLoyaltyRuleCheckerSpec.php]: missing blankline

**Comment by pjedrzejewski** [comment]
Same as #3305, let's see what will happen with #3368.

**Pull request closed without merging**